record_tx.go

```diff
@@ -0,0 +1,98 @@
+package bux
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/libsv/go-bt"
+)
+
+type recordTxStrategy interface {
+	TxID() string
+	Validate() error
+	Execute(ctx context.Context, c ClientInterface, opts []ModelOps) (*Transaction, error)
+}
+
+type recordIncomingTxStrategy interface {
+	ForceBroadcast(force bool)
+}
+
+func recordTransaction(ctx context.Context, c ClientInterface, strategy recordTxStrategy, opts ...ModelOps) (*Transaction, error) {
+	unlock := waitForRecordTxWriteLock(ctx, c, strategy.TxID())
+	defer unlock()
+
+	transaction, err := strategy.Execute(ctx, c, opts)
+	return transaction, err
+}
+
+func getRecordTxStrategy(ctx context.Context, c ClientInterface, xPubKey, txHex, draftID string) (recordTxStrategy, error) {
+	var rts recordTxStrategy
+
+	if draftID != "" {
+		rts = &outgoingTx{
+			Hex:            txHex,
+			RelatedDraftID: draftID,
+			XPubKey:        xPubKey,
+		}
+	} else {
+		tx, err := getTransactionByHex(ctx, c, txHex)
```

Comment:
For me to make this code more self descriptive it would be nice if in this if/else (especially in else) create a smaller method/function to handle this code and it's name should tell us what it is exactly doing.

Reply:
I changed it... TBH I don't think it's better now but have no idea how make it better